web/src/project-activity.js

```diff
@@ -93,6 +93,73 @@ class ProjectActivity extends Component {
     return(content);
   };
 
+  handleUpdateAllAction = (projectList) => {
+    const modalCompletion = choice => {
+      if (choice === 'ok') {
+        this.props.updateAllProjects(projectList,
+          successArr => {
+            successArr = successArr.sort((a, b) => a.name < b.name ? -1 : a.name > b.name ? 1 : 0);
+            this.props.getProjectSummary();
+            this.props.refreshCodeDir();
+            this.props.showModal(<ModalContent
+              message='Updating all projects succeeded.'
+              supporting={`${successArr.length ? `The following projects were updated:\n\n${successArr.map(e => e.name).join('\n')}.` : ''}`}
+              style={{whiteSpace: 'pre-line'}}
+              buttonAction={this.modalDismiss}
+              confirmOnly={true}
+            />)
+          },
+          (successArr, failureArr) => {
+            successArr = successArr.sort((a, b) => a.name < b.name ? -1 : a.name > b.name ? 1 : 0);
+            failureArr = failureArr.sort((a, b) => a.name < b.name ? -1 : a.name > b.name ? 1 : 0);
+            this.props.getProjectSummary();
+            this.props.refreshCodeDir();
+            this.props.showModal(<ModalContent
+              message='Updating all projects failed.'
+              supporting={<span>{successArr.length ? (<div>The following projects were updated:<br /><br /><table><tbody>{successArr.map(e => (<tr><td>{e.name}</td></tr>))}</tbody></table><br /><br /></div>) : ''}
+              <div>The following errors happened:<br /><br /><table><tbody>{failureArr.map(e => (<tr><td style={{ width: '100px' }}>{e.name}</td><td style={{ width: 'auto' }}>{e.failureResult.error}</td></tr>))}</tbody></table></div></span>}
+              buttonAction={this.modalDismiss}
+              confirmOnly={true}
+            />)
+          });
+        this.props.showModal(
+          <ModalContent
+            message='Updating all projects'
+            supporting="please wait..."
+            buttonAction={this.modalDismiss}
+            confirmOnly={true}
+          />
+        );
+      } else {
+        // update request canceled
+        this.props.hideModal();
+        <table>
+    <thead>
+        <tr>
+            <th colspan="2">The table header</th>
+        </tr>
+    </thead>
```

Comment:
@jlmitch5 - this stray table markup seems like a left over from some earlier work, correct?